examples/bert/bert_finetune_glue.py

```diff
@@ -270,9 +296,57 @@ def preprocess_data(inputs, labels):
         f"The best hyperparameters found are:\nLearning Rate: {best_hp['lr']}"
     )
 
-    if FLAGS.do_evaluation:
-        print("Evaluating on test set.")
-        finetuning_model.evaluate(test_ds)
+    if FLAGS.tsv_prediction_output:
+        filenames = {
+            "cola": "CoLA.tsv",
+            "sst2": "SST-2.tsv",
+            "mrpc": "MRPC.tsv",
+            "qqp": "QQP.tsv",
+            "stsb": "STS-B.tsv",
+            "mnli_mismatched": "MNLI-mm.tsv",
+            "mnli_matched": "MNLI-m.tsv",
+            "qnli": "QNLI.tsv",
+            "rte": "RTE.tsv",
+            "wnli": "WNLI.tsv",
+        }
+
+        labelnames = {
+            "mnli_matched": ["entailment", "neutral", "contradiction"],
+            "mnli_mismatched": ["entailment", "neutral", "contradiction"],
+            "qnli": ["entailment", "not_entailment"],
+            "rte": ["entailment", "not_entailment"],
+        }
+
+        filename = (
+            FLAGS.tsv_prediction_output + "/" + filenames[FLAGS.task_name]
+        )
+
+        @tf.function
+        def eval_step(inputs):
+            def step_fn(inputs):
+                prob = finetuning_model(inputs)
+                pred = tf.argmax(prob, -1)
+                return pred
+
+            return strategy.run(step_fn, args=(inputs,))
+
+        labelname = labelnames.get(FLAGS.task_name)
+        with tf.io.gfile.GFile(filename, "w") as f:
+            # Write the required headline for GLUE.
```

Comment:
I think you can use the builtin csv with a different delimiter to write tsvs
https://docs.python.org/3/library/csv.html
https://stackoverflow.com/q/29895602

We should prefer that over doing this writing outself

Reply:
done